src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws AbsentDaoAnnotationException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoSuchMethodException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getType().equals(BetDao.class)) {
+                if (!Factory.getBetDao().getClass().isAnnotationPresent(Dao.class)) {
+                    throw new AbsentDaoAnnotationException("Dao annotation is absent");
+                }
+                field.setAccessible(true);
+                field.set(instance, Factory.getBetDao());
+            }
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getType().equals(UserDao.class)) {
+                if (!Factory.getUserDao().getClass().isAnnotationPresent(Dao.class)) {
+                    throw new AbsentDaoAnnotationException("Dao annotation is absent");
+                }
+                field.setAccessible(true);
+                field.set(instance, Factory.getUserDao());
+            }
```

Comment:
let's try to simplify this part

let's have first `if` checking if field has `Inject` annotation

then inside that `if`:

1. check if the field's type is BetDao and there is `@Dao` in BetDaoImpl - then set field
2. else - check if the field's type is UserDao and there is `@Dao` in UserDaoImpl - then set field
3. else - throw Exception